src/astro/sql/operators/agnostic_load_file.py

```diff
@@ -132,16 +135,66 @@ def _load_dataframe(self, path):
         with smart_open.open(
             path, mode=mode.get(file_type, "r"), transport_params=transport_params
         ) as stream:
-            return deserialiser[file_type](
-                stream, **deserialiser_params.get(file_type, {})
-            )
+            if file_type == "ndjson":
+                df = None
+                count = 0
+                while True:
+                    rows = stream.readlines(DEFAULT_CHUNK_SIZE)
+                    if len(rows) == 0:
+                        break
+                    if df is None:
+                        df = pd.DataFrame(self.process_ndjson(rows))
+                    else:
+                        df = df.append(pd.DataFrame(self.process_ndjson(rows)))
+                    print("count : ", count)
+                    count += 1
+                return df
+            else:
+                return deserialiser[file_type](
+                    stream, **deserialiser_params.get(file_type, {})
+                )
+
+    def process_ndjson(self, rows):
+        results = {}
+        for row in rows:
+
+            row = json.loads(row)
+            if len(row) == 0 or len(row.keys()) != 8:
+                continue
+
+            for k in row:
+                if not results.get(k):
+                    results[k] = []
+
+                if k in self.flatten_ndjson:
+                    v = self.flatten_ndjson[k]
+
+                    if isinstance(v, str):
+                        results[k].append(self.ndjson_get_values(v, row))
+                    elif self.isLambda(v):
+                        results[k].append(v(row))
```

Comment:
what does isLambda mean?

Reply:
Have removed this from code, but the idea was to give access to entire row within a lambda function. So user can do operation like creating new col by function execution.